### DISCUSSION TITLE
Databank post

### DISCUSSION THREAD

**Discussion by WalterRestifo** [neuefische/questions#408] [Web]
I want to save in my databank an object that has a key containing an array of strings. This is my code now, but it gives me a "Bad request " error. My schema looks like this: 
<img width="1599" alt="Bildschirmfoto 2022-12-16 um 13 00 28" src="https://user-images.githubusercontent.com/116001970/208093833-160c3ae5-b785-4e4a-ae49-ac2522047f17.png">
I tried also to code {type : [String], required: true} on 8

This is my post function:
<img width="1327" alt="Bildschirmfoto 2022-12-16 um 13 04 44" src="https://user-images.githubusercontent.com/116001970/208094390-47c126ce-593c-440f-a076-80ff46f9ea8f.png">

What could be the problem?

**Comment by mihaimazareanu** [comment]
Hi Walter,

It seems you forgot to specify where you would like to save the data sent with this model

So in your model, in row 12, your `const AllPlayers` should look like this:

    const AllPlayers =
        mongoos.models.AllPlayers || mongoose.model("AllPlayers", allPlayersSchema, "name of your collection in your DB");

And in line 8, you should use the same syntax as for the other elements: 

    language: { type: String },

You know if you need that element should be required or not, if yes, then you should add `required: true` in line 8.
        
I hope this helps.

Also, for the future, please post your code directly and not a photo of our code. Thanks. 🙏

**Reply by WalterRestifo** [reply]
Hi, I found the problem. The schema is good as it is, I had a bug in my function that gives the value to skill and gender, because the function used to give the value of the language key use a state and state changes. After a state change the whole code get rerendered and the object ownPlayerCard take his default value (that is gender = "" and skill = ""). Gender and skill are required and that is why I had a bad request. Now it works, but thank you for your help! 🙂
